### PR TITLE
TD Info Model fixes

### PR DESCRIPTION
**Pull request by sebastiankb**
fix securityDefinitions from #1324

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/pull/1334.html" title="Last updated on Jan 12, 2022, 3:51 PM UTC (fc83764)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/1334/21ce84a...fc83764.html" title="Last updated on Jan 12, 2022, 3:51 PM UTC (fc83764)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1201,15 +1201,15 @@ <h2>Core Vocabulary Definitions</h2>
...
+          <code>https://www.w3.org/2019/wot/td/v1</code> either
           directly when of type <code>anyURI</code> or as first
           element when of type <a href="#dfn-array" class=
           "internalDFN" data-link-type="dfn">Array</a>.</span>
           <span class="rfc2119-assertion" id=
           "td-context-ns-thing-optional">When <code>@context</code>
           is an <a href="#dfn-array" class="internalDFN"
           data-link-type="dfn">Array</a>, the anyURI
-          <code>https://www.w3.org/2022/wot/td/v1.1</code> <em class=
+          <code>https://www.w3.org/2019/wot/td/v1</code> <em class=
```
Comment (relu91) [inline index.html:1212]: Are these changes intended? cause in https://github.com/w3c/wot-thing-description/pull/1333 there are the same changes but from 2019 to 2022 🤔

Comment (sebastiankb) [reply index.html]: many thanks, I fixed it.

File: index.html [hunk 2]
```diff
@@ -2039,7 +2051,7 @@ <h2>Hypermedia Controls Vocabulary Definitions</h2>
                 <td>
                     0..*
                 </td>
-                <td>Point to a software implementation of the TD.</td>
+                <td>Point to the web app manifest of a web application which provides a user interface with which a user can interact with the Thing (also see [[APPMANIFEST]]).</td>
```
Comment (mlagally) [inline index.html:2054]: This is a significant modification of the previous text.

These changes introduce a specific format to describe Web applications, which restricts the UI components to WebApps. Is this intended? 

The referenced document is a working draft, seems not to be suitable for a normative reference.

These late changes are difficult to review, and the implications cannot be properly considered in the limited review time.

File: index.html [hunk 3]
```diff
@@ -2395,7 +2407,13 @@ <h2>Hypermedia Controls Vocabulary Definitions</h2>
...
                 (e.g., <code>text/plain</code>) and potential
                 parameters (e.g., <code>charset=utf-8</code>) for
-                the media type [<cite><a class="bibref" data-link-type="biblio" href="#bib-rfc2046" title="Multipurpose Internet Mail Extensions (MIME) Part Two: Media Types">RFC2046</a></cite>].</td><td>mandatory</td><td><a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#string"><code>string</code></a></td></tr></tbody></table></section>
+                the media type [<cite><a class="bibref" data-link-type="biblio" href="#bib-rfc2046" title="Multipurpose Internet Mail Extensions (MIME) Part Two: Media Types">RFC2046</a></cite>].</td><td>mandatory</td><td><a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#string"><code>string</code></a></td></tr>
+<tr class="rfc2119-table-assertion" id="td-vocab-schema--ExpectedResponse"><td><code>schema</code></td><td>Used to define the output data schema
```
Comment (mlagally) [inline index.html:2411]: Please clarify why this normative assertion is added.

File: validation/td-validation.ttl [hunk 4]
```diff
@@ -1082,7 +1094,7 @@
                 <td>
                     0..*
                 </td>
-                <td>Point to a software implementation of the TD.</td>
+                <td>Point to the web app manifest of a web application which provides a user interface with which a user can interact with the Thing (also see [[APPMANIFEST]]).</td>
```
Comment (mlagally) [inline validation/td-validation.ttl:1097]: See previous comment. This introduces a new specification, which is a working draft.

File: index.html (hunk 3 above)
Comment (mlagally) [reply index.html:2411]: These changes introduce new normative behavior and may have undesired impact.
They should be reviewed and discussed in the group before they are added.

**Review by mlagally — CHANGES_REQUESTED** [review]
This merge request contains additional normative assertions.

These come very late, and Oracle needs more time to properly review them and consider the implications.

This MR should not be merged without careful review and further discussion.

File: index.html [hunk 5]
```diff
@@ -1189,22 +1189,33 @@ <h2>Core Vocabulary Definitions</h2>
...
+          The TD 1.1 context may exceptionally be placed as the second <a href="#dfn-array" class="internalDFN"
+          data-link-type="dfn">Array</a> entry if the TD 1.0 context must be specified as the first entry to allow 
+          backward compatibility. 
+          <p class="note">
+           A Thing Description can use the 1.0 context (if a valid 1.0 TD according to [[WoT-Thing-Description]]), 1.1 
+           context (if a valid 1.1 TD according to this specification) or both (if it's both a valid 1.0 TD and 1.1 TD). 
+           A 1.0 consumer can only be expected to parse a Thing Description which includes 
+           the 1.0 context, but can not be expected to parse a Thing Description which 
+           only has a 1.1 context. A Thing Description which only contains 
+           the 1.1 context should be a valid 1.1 Thing Description.
```
[the remark this answers is not in the document]
Comment (sebastiankb) [reply index.html:1217]: no, just a note.

File: index.html (hunk 2 above)
Comment (sebastiankb) [reply index.html:2054]: Please see #1324 

image

File: index.html (hunk 3 above)
Comment (sebastiankb) [reply index.html:2411]: see #1324 and https://github.com/w3c/wot-thing-description/issues/1053#issuecomment-959452562

File: index.html [hunk 6]
```diff
@@ -1189,22 +1189,33 @@ <h2>Core Vocabulary Definitions</h2>
...
+                <code>ExpectedResponse</code> and <code>AdditionalExpectedResponse</code> object.</td><td>optional</td><td><a>Map</a> of <a href="#dataschema"><code>DataSchema</code></a></td></tr></tbody></table><p><span class="rfc2119-assertion" id=
           "td-context-ns-thing-mandatory">The <code>@context</code>
           name-value pair <em class="rfc2119" title=
           "MUST">MUST</em> contain the anyURI
           <code>https://www.w3.org/2022/wot/td/v1.1</code> either
           directly when of type <code>anyURI</code> or as first
           element when of type <a href="#dfn-array" class=
           "internalDFN" data-link-type="dfn">Array</a>.</span>
+          The TD 1.1 context may exceptionally be placed as the second <a href="#dfn-array" class="internalDFN"
+          data-link-type="dfn">Array</a> entry if the TD 1.0 context must be specified as the first entry to allow
```
[the remark this answers is not in the document]
Comment (sebastiankb) [reply index.html:1209]: see #1324 and https://github.com/w3c/wot-thing-description/issues/1243#issuecomment-949600882

File: index.html [hunk 7]
```diff
@@ -1189,22 +1189,33 @@ <h2>Core Vocabulary Definitions</h2>
...
+                name-value pair such as inside an 
+                <code>ExpectedResponse</code> and <code>AdditionalExpectedResponse</code> object.</td><td>optional</td><td><a>Map</a> of <a href="#dataschema"><code>DataSchema</code></a></td></tr></tbody></table><p><span class="rfc2119-assertion" id=
           "td-context-ns-thing-mandatory">The <code>@context</code>
           name-value pair <em class="rfc2119" title=
           "MUST">MUST</em> contain the anyURI
           <code>https://www.w3.org/2022/wot/td/v1.1</code> either
           directly when of type <code>anyURI</code> or as first
           element when of type <a href="#dfn-array" class=
           "internalDFN" data-link-type="dfn">Array</a>.</span>
+          The TD 1.1 context may exceptionally be placed as the second <a href="#dfn-array" class="internalDFN"
```
[the remark this answers is not in the document]
Comment (sebastiankb) [reply index.html:1208]: This should help to make clear that a TD that follows this specification should have also the TD 1.0 context if TD 1.0 processor are used. As next there is also a note about this situation.

[the remark this answers is not in the document]
Comment (sebastiankb) [reply index.html]: this text and the note is removed again and a separate PR was provided #1339

File: index.html (hunk 2 above)
Comment (sebastiankb) [reply index.html]: I reverted this as well and put this in a separate PR #1340

File: validation/td-validation.ttl (hunk 4 above)
Comment (sebastiankb) [reply validation/td-validation.ttl]: is reverted. see comment above

File: index.html (hunk 3 above)
Comment (sebastiankb) [reply index.html]: this is also reverted and outsourced in a separate PR #1341

**Comment by sebastiankb** [comment]
@mlagally many thanks for reviewing this PR. This PR addresses the feedback given in #1324.

I outsourced your related comments in separate PRs #1341, #1340 and #1339 which we can be discussed and decided in tomorrow's call.

**Comment by sebastiankb** [comment]
from today's TD call, decided to merge
